cdk-app/lib/cdk-app-stack.ts

```diff
@@ -0,0 +1,87 @@
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as cloudfront_origins from 'aws-cdk-lib/aws-cloudfront-origins';
+import * as s3deploy from 'aws-cdk-lib/aws-s3-deployment';
+
+import { BlockPublicAccess, BucketAccessControl } from 'aws-cdk-lib/aws-s3';
+
+import {
+  CfnOutput,
+  Duration,
+  RemovalPolicy,
+  Stack,
+  StackProps,
+} from 'aws-cdk-lib';
+
+// import * as sqs from 'aws-cdk-lib/aws-sqs';
+
+export class CdkAppStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // The code that defines your stack goes here
+
+    // Create an S3 bucket to store content and set removal policy to eithrt 'Retain' or 'Destroy'
+     const siteBucket = new s3.Bucket(this, 'SiteBucket', {
+       bucketName: 'rsschool-cdk-deployment-bucket',
+       publicReadAccess: true,
```

Comment:
According to the task you need to deny access to your site through S3-link. So you should set the `publicReadAccess` prop to `false` and use it in combination with `blockPublicAccess` rule.

Reply:
Set `publicReadAccess` to `false` and also changed `blockPublicAccess: BlockPublicAccess.BLOCK_ALL` to `blockPublicAccess: BlockPublicAccess.BLOCK_ALL` to avoid accessing the object due to caching or the session still being active.
``` TypeScript  
     const siteBucket = new s3.Bucket(this, 'SiteBucket', {
       publicReadAccess: false,
       blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
     });
```